releaseinfo/registryclient_provider.go

```diff
@@ -0,0 +1,41 @@
+package releaseinfo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/hypershift/releaseinfo/registryclient"
+)
+
+const (
+	ReleaseImageStreamFile   = "release-manifests/image-references"
+	ReleaseImageMetadataFile = "release-manifests/0000_50_installer_coreos-bootimages.yaml"
+)
+
+var _ Provider = (*RegistryClientProvider)(nil)
+
+// RegistryClientProvider uses a registry client to directly stream image
+// content and extract image metadata.
+type RegistryClientProvider struct {
+}
+
+func (p *RegistryClientProvider) Lookup(ctx context.Context, image string, pullSecret []byte) (releaseImage *ReleaseImage, err error) {
+	fileContents, err := registryclient.ExtractImageFiles(ctx, image, pullSecret, ReleaseImageStreamFile, ReleaseImageMetadataFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract release metadata: %w", err)
+	}
+
+	imageStream, err := DeserializeImageStream(fileContents[ReleaseImageStreamFile])
+	if err != nil {
+		return nil, err
+	}
+
+	coreOSMeta, err := DeserializeImageMetadata(fileContents[ReleaseImageMetadataFile])
```

Comment:
Should validate the key exists and throw a meaningful error otherwise

Reply:
Added check